### PR TITLE
Updating dep versions for bh2 install

### PR DESCRIPTION
**Pull request by Andrewq11**
## Changelogs

- Pinning version of `numpy` to the latest version < 2.0. `graphium` uses the old `numpy` API and must be below version 2.0 for now.
- Unpinning the version for the `cuda-version` package. This allows for the CUDA-compatible version of `pytorch` to be installed
- Pinning the version of `rdkit` to a recent version which contains all of the required files (imports) for `graphium_cpp`

---

_Checklist:_

- [ ] ~_Was this PR discussed in an issue? It is recommended to first discuss a new feature into a GitHub issue before opening a PR._~
- [ ] ~_Add tests to cover the fixed bug(s) or the new introduced feature(s) (if appropriate)._~
- [ ] ~_Update the API documentation is a new function is added, or an existing one is deleted._~
- [x] _Write concise and explanatory changelogs above._
- [x] _If possible, assign one of the following labels to the PR: `feature`, `fix` or `test` (or ask a maintainer to do it for you)._

---

_discussion related to that PR_

### CODE REVIEW

File: env.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ dependencies:
   - pytorch_scatter >=2.0
 
   # chemistry
-  - rdkit
+  - rdkit == 2024.03.4
```
Comment (DomInvivo) [inline env.yml:44]: Why?? And why not `<= 2024.03.3`?

**Review by DomInvivo — COMMENTED** [review]
Just a comment about rdkit

File: env.yml (hunk 1 above)
Comment (Andrewq11) [reply env.yml:44]: The latest version of `rdkit` was not being installed with the required files that are imported by `graphium_cpp`. So, I pinned it to a very recent version. We could do `<= 2024.03.4`, but can we confirm that all earlier versions of `rdkit` contain the required files needed by `graphium_cpp`?

Comment (Andrewq11) [reply env.yml:44]: Let me also clean my `conda` cache and confirm that it wasn't just a local cache issue with the latest version of `rdkit`

Comment (Andrewq11) [reply env.yml:44]: Confirmed that the latest version of `rdkit` on `conda` does not include the required files for `graphium_cpp`

Comment (DomInvivo) [reply env.yml:44]: Yeah this was something that @ndickson-nvidia mentioned as well.

I will merge the PR

**Pull request merged**